### PR TITLE
Document optional fields & arguments

### PR DESCRIPTION
**Pull request by badboy**
While playing around with UniFFI, in order to figure out how we can use it in Glean, I came across these essential things for us: default values.

While implemented it lacks documentation, so I went ahead and added that.

### CODE REVIEW

File: docs/manual/src/udl/structs.md [hunk 1]
```diff
@@ -57,9 +57,78 @@ struct TodoEntry {
...
+    // ...
+}
+```
+
+This works for Swift and Python targets too.
+If not set otherwise the default value for a field is passed to the Rust constructor.
+
+## Optional fields and default values
+
+Fields can be made optional using a `T?` type.
```
Comment (badboy) [inline docs/manual/src/udl/structs.md:94]: I could be convinced that this is not necessary, as the `T?` thing is already document in Types and thus having default values for it with `null` is covered by the above paragraph.
I still figured it's worth calling out specifically here.

Just let me know what you think, I can change that.

**Comment by audkar** [comment]
Nice! Just  found out about `uniffi` today, and was looking at default values myself. What a coincident. 

Also would be nice to have working example showcasing this functionality. In some cases (if CI includes examples) code example are easier to maintain then static docs which can go out-of-date and bring more frustration then benefits to users...

**Comment by badboy** [comment]
> Also would be nice to have working example showcasing this functionality. In some cases (if CI includes examples) code example are easier to maintain then static docs which can go out-of-date and bring more frustration then benefits to users...

Sure, but let's start simple

File: docs/manual/src/udl/structs.md (hunk 1 above)
Comment (mhammond) [reply docs/manual/src/udl/structs.md:94]: I don't think it hurts to mention it here too. However, I suggest maybe adding what the rust receiver looks like too (ie, takes an Option<T>), just like you've done above for the defaults.

**Review by mhammond — COMMENTED** [review]
This looks great to me, but we might as well wait a few days for Ryan to weigh in.

File: docs/manual/src/udl/structs.md (hunk 1 above)
Comment (rfk) [reply docs/manual/src/udl/structs.md:94]: I agree it seems worth calling out here as well, it may be easier to understand with these concrete examples.

**Review by rfk — APPROVED** [review]
Thanks for diving in! I'll be excited to hear how your explorations went for Glean, whether positive or negative :-)

**Pull request merged**